src/SOS/Strike/platform/targetimpl.cpp

```diff
@@ -0,0 +1,350 @@
+    {
+        s_target = new Target();
+        OnUnloadTask::Register(CleanupTarget);
+    }
+    s_target->AddRef();
+    return s_target;
+}
+
+/**********************************************************************\
+ * Creates an instance of the runtime class. First it attempts to create
+ * the .NET Core instance and if that fails, it will try to create the
+ * desktop CLR instance.  If both runtimes exists in the process or dump
+ * this runtime only creates the .NET Core version and leaves creating
+ * the desktop instance on demand in SwitchRuntime.
+\**********************************************************************/
+HRESULT 
+Target::CreateInstance(IRuntime **ppRuntime)
+{
+    HRESULT hr = S_OK;
+    if (*ppRuntime == nullptr)
+    {
+        hr = Runtime::CreateInstance(this, IRuntime::Core, &m_netcore);
+#ifdef FEATURE_PAL
+        *ppRuntime = m_netcore;
+#else
+        ITarget::OperatingSystem os = this->GetOperatingSystem();
+        if (FAILED(hr))
+        {
+            if (os == ITarget::OperatingSystem::Linux || os == ITarget::OperatingSystem::OSX)
+            {
+                hr = Runtime::CreateInstance(this, IRuntime::UnixCore, &m_netcore);
+            }
+        }
+        if (FAILED(hr))
+        {
+            if (os == ITarget::OperatingSystem::Windows)
+            {
+                hr = Runtime::CreateInstance(this, IRuntime::WindowsDesktop, &m_desktop);
+            }
+        }
+        *ppRuntime = m_netcore != nullptr ? m_netcore : m_desktop;
+#endif
+    }
+    return hr;
+}
+
+/**********************************************************************\
+ * Switches between the .NET Core and desktop runtimes (if both 
+ * loaded). Creates the desktop CLR runtime instance on demand.
+\**********************************************************************/
+#ifndef FEATURE_PAL
+bool Target::SwitchRuntimeInstance(bool desktop)
+{
+    if (desktop) 
+    {
+        if (this->GetOperatingSystem() != ITarget::OperatingSystem::Windows) {
+            return false;
+        }
+        Runtime::CreateInstance(this, IRuntime::WindowsDesktop, &m_desktop);
+    }
+    IRuntime* runtime = desktop ? m_desktop : m_netcore;
+    if (runtime == nullptr) {
+        return false;
+    }
+    g_pRuntime = runtime;
+    return true;
+}
+#endif
+
+/**********************************************************************\
+ * Set the runtime directory path
+\**********************************************************************/
+void Target::SetRuntimeDirectoryInstance(LPCSTR runtimeModulePath)
+{
+    if (m_runtimeModulePath != nullptr) {
+        free((void*)m_runtimeModulePath);
+    }
+    m_runtimeModulePath = _strdup(runtimeModulePath);
+}
+
+/**********************************************************************\
+ * Display the internal target and runtime status
+\**********************************************************************/
+void Target::DisplayStatusInstance()
+{
+    static const char* osName[] = {
```

Comment:
We should probably move this next to the `ITarget::OperatingSystem` definition to ensure they always match.

Reply:
The target.h/ITarget interface is the abstraction. I don't think implementation details like this belong there.